src/httpClient.ts

```diff
@@ -151,3 +152,113 @@ export class HttpClientResponse<T> implements IHttpClientResponse {
         this.body = body;
     }
 }
+
+export class HttpApiClient extends HttpClient {
+
+    baseUrl: any;
+
+    constructor(baseUrl: string = undefined) {
+        super();
+        this.baseUrl = baseUrl;
```

Comment:
Solved by setting configuration in constructor

    constructor(baseUrl: string = undefined) {

        super();

        this.configure((config) => {
            if (baseUrl) {
                config.withBaseUrl(baseUrl);
            }
            this.configuration = config;
        });

    }